### PR TITLE
Fixes #29525 - Use model name from global id to load object

### PR DESCRIPTION
**Pull request by kamils-iRonin**
we can use the model name encoded in GlobalId to load the object

### CODE REVIEW

File: app/graphql/foreman_graphql_schema.rb [hunk 1]
```diff
@@ -19,10 +19,8 @@ def self.id_from_object(object, type_definition, query_ctx)
   def self.object_from_id(id, query_ctx)
     return unless id.present?
 
-    _, type_name, item_id = Foreman::GlobalId.decode(id)
-    type_class = "::Types::#{type_name}".safe_constantize
```
Comment (kamils-iRonin) [inline app/graphql/foreman_graphql_schema.rb:23]: it won't recognize the type from the plugin namespace, eg. `ForemanMyPlugin::Types::MyType`

**Comment by mmoll** [comment]
[test katello]

File: app/graphql/foreman_graphql_schema.rb (hunk 1 above)
Comment (timogoebel) [reply app/graphql/foreman_graphql_schema.rb:23]: I believe this was added as a security feature. The global id is user input and I believe it's not recommended to call `constantize` on user input. Can we maybe filter this to only allow classes that are Types?

Comment (kamils-iRonin) [reply app/graphql/foreman_graphql_schema.rb:23]: good catch, what do you think about `ForemanGraphqlSchema.types.keys.find { |t| t == type_name }&.safe_constantize`?

Comment (timogoebel) [reply app/graphql/foreman_graphql_schema.rb:23]: This looks good.

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
Thanks, @kamils-iRonin.

**Comment by xprazak2** [comment]
Do you have namespaced models in plugins? I stumbled on a funny case where `ForemanGrapqlSchema::object_from_id` tries to look for `Policy` instead of `ForemanOpenscap::Policy`, so I am wondering if I have the type set up correctly...

**Comment by kamils-iRonin** [comment]
> Do you have namespaced models in plugins? I stumbled on a funny case where `ForemanGrapqlSchema::object_from_id` tries to look for `Policy` instead of `ForemanOpenscap::Policy`, so I am wondering if I have the type set up correctly...

@xprazak2 have you configured `graphql_type` in your model? like here 

https://github.com/theforeman/foreman/blob/c70f81b98043ad158635c5fdff77891581bc9631/app/models/host/managed.rb#L105

**Comment by timogoebel** [comment]
@xprazak2: Other magic we sometimes use looks like this (authored by @kamils-iRonin):

```ruby
# frozen_string_literal: true

module ForemanGafferTape
  module Types
    class Application < ::Types::BaseObject
      model_class ForemanGafferTape::Application

      description 'An Application'

      global_id_field :id
      timestamps
      field :name, String

      has_many :systemgroups, Types::Systemgroup

      def self.graphql_definition
        super.tap { |type| type.instance_variable_set(:@name, 'ForemanGafferTape::Application') }
      end
    end
  end
end
```

**Comment by xprazak2** [comment]
I have `graphql_type` but it does not have a desired effect, `graphql_definition` method worked perfectly. Thanks!

**Pull request merged**